### PR TITLE
Add whoami audit integration test

### PR DESCRIPTION
**Pull request by amosmintzcyberark**
### What does this PR do?
- Added integration test (cucumber) for whoami command audit

### What ticket does this PR close?
add more tests pass PR: https://github.com/cyberark/conjur/pull/2061 following ticket:https://github.com/cyberark/conjur/issues/2052 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [X] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [X] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [X] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: app/controllers/status_controller.rb [hunk 1]
```diff
@@ -16,13 +17,14 @@ def index
...
-
     render json: {
       client_ip: request.ip,
       user_agent: request.user_agent,
       account: token_user.account,
       username: token_user.login,
-      token_issued_at: Time.at(token_user.token.claims["iat"])
+      # This is here to fix a bug that occurs only in test
+      # We don't fully understand what causes the bug but this is the hack we settled on to fix it.
+      token_issued_at: ((defined? token_user.token.claims) == nil) ? nil : Time.at(token_user.token.claims["iat"])
```
Comment (orenbm) [inline app/controllers/status_controller.rb:27]: can't we write 
```
token_issued_at: token_user.token.claims.nil? ? nil : Time.at(token_user.token.claims["iat"])
```

File: cucumber/api/features/whoami.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    And there is an audit record matching:
+    """
+      <38>1 * * conjur * identity-check
+      [subject@43868 role="cucumber:user:alice"]
+      [auth@43868 user="cucumber:user:alice"]
+      [client@43868 ip="*"]
+      [action@43868 result="success" operation="check"]
+      * *
+      cucumber:user:alice checked its identity using whoami
+    """
```
Comment (orenbm) [inline cucumber/api/features/whoami.feature:16]: missing new line

File: cucumber/api/features/whoami.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Feature: Who Am I
+
+  Scenario: Audit entry
+
+    Given I am a user named "alice"
+    And I can GET "/whoami"
```
Comment (orenbm) [inline cucumber/api/features/whoami.feature:6]: this looks like a `When` (and also remove the `can`)

File: cucumber/api/features/whoami.feature [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    And there is an audit record matching:
```
Comment (orenbm) [inline cucumber/api/features/whoami.feature:7]: this looks like a `Then`

File: app/controllers/status_controller.rb (hunk 1 above)
Comment (orenbm) [reply app/controllers/status_controller.rb:27]: can you elaborate a bit on the bug?

Comment (amosmintzcyberark) [reply app/controllers/status_controller.rb:27]: The method '**claims**' doesn't exist when I run the test. I tried to debug it and change the token on the sender side (rest_helpers.rb in cucumber and other places using Slosilo). but, I couldn't find the cause.

File: cucumber/api/features/whoami.feature (hunk 3 above)
Comment (amosmintzcyberark) [reply cucumber/api/features/whoami.feature:6]: I'll replace the 'can' to 'successfully'... what do you think ?

File: cucumber/api/features/whoami.feature (hunk 2 above)
Comment (amosmintzcyberark) [reply cucumber/api/features/whoami.feature:16]: ok

File: cucumber/api/features/whoami.feature (hunk 4 above)
Comment (amosmintzcyberark) [reply cucumber/api/features/whoami.feature]: ok

File: app/controllers/status_controller.rb (hunk 1 above)
Comment (orenbm) [reply app/controllers/status_controller.rb:27]: please look into it some more. it is best not to change application code for the purposes of tests.

File: cucumber/api/features/support/rest_helpers.rb [hunk 5]
```diff
@@ -282,7 +282,10 @@ def current_user_api_key
 
   def current_user_credentials
     username = @current_user.login
-    token = Slosilo["authn:#{@current_user.account}"].signed_token(username)
+    # Configure Slosilo to produce valid access tokens
+    slosilo = Slosilo["authn:#{@current_user.account}"] ||= Slosilo::Key.new
```
Comment (uCatu) [inline cucumber/api/features/support/rest_helpers.rb:286]: @amosmintzcyberark 
Can we add test to a case where `cucumber:user:alice` is not present at Slosilo leading us to  `Slosilo::Key.new`

**Review by uCatu — CHANGES_REQUESTED** [review]

File: cucumber/api/features/support/rest_helpers.rb (hunk 5 above)
Comment (amosmintzcyberark) [reply cucumber/api/features/support/rest_helpers.rb:286]: we can, but it's not related to the whoami, it's related to authentication. the whoami can't fail on authentication (it's a paradox)

**Review by uCatu — APPROVED** [review]

**Pull request merged**